RecoCTPPS/TotemRPLocal/src/TotemTimingRecHitProducerAlgorithm.cc

```diff
@@ -160,37 +156,27 @@ int TotemTimingRecHitProducerAlgorithm::fastDiscriminator(
   return threholdCrossingIndex;
 }
 
-float TotemTimingRecHitProducerAlgorithm::constantFractionDiscriminator(
-    const std::vector<float> &time, const std::vector<float> &data) {
+float
+TotemTimingRecHitProducerAlgorithm::constantFractionDiscriminator(const std::vector<float>& time, const std::vector<float>& data)
+{
   std::vector<float> dataProcessed(data);
-  if (lowPassFrequency_ != 0) {
-    // Smoothing
-    for (int i = 0; i < (int)data.size(); ++i) {
-      for (int j = -smoothingPoints_ / 2;
-           j <= +smoothingPoints_ / 2; ++j) {
-        if ((i + j) >= 0 && (i + j) < (int)data.size() && j != 0) {
+  if (lowPassFrequency_ != 0) // Smoothing
+    for (unsigned short i = 0; i < data.size(); ++i)
+      for (unsigned short j = -smoothingPoints_/2; j <= +smoothingPoints_/2; ++j)
+        if ((i+j) >= 0 && (i+j) < data.size() && j != 0) {
           float x = SINC_COEFFICIENT * lowPassFrequency_ * j;
           dataProcessed[i] += data[i + j] * std::sin(x) / x;
         }
-      }
-    }
-  }
   auto max_it = std::max_element(dataProcessed.begin(), dataProcessed.end());
   float max = *max_it;
 
   float threshold = cfdFraction_ * max;
   int indexOfThresholdCrossing = fastDiscriminator(dataProcessed, threshold);
 
-  float t = TotemTimingRecHit::NO_T_AVAILABLE;
-  if (indexOfThresholdCrossing >= baselinePoints_ &&
-      indexOfThresholdCrossing < (int)time.size()) {
-    t = (time[indexOfThresholdCrossing - 1] -
-         time[indexOfThresholdCrossing]) /
-            (dataProcessed[indexOfThresholdCrossing - 1] -
-             dataProcessed[indexOfThresholdCrossing]) *
-            (threshold - dataProcessed[indexOfThresholdCrossing]) +
-        time[indexOfThresholdCrossing];
-  }
-
-  return t;
+  return (indexOfThresholdCrossing >= baselinePoints_ && indexOfThresholdCrossing < (int)time.size())
+    ? (time[indexOfThresholdCrossing-1]-time[indexOfThresholdCrossing])
```

Comment:
is it obvious here that indexOfThresholdCrossing is also ``` >=1``` ?
Current config is OK (```baselinePoints``` is set to 8), but better check explicitly.

Reply:
Extra check introduced in the ternary operator condition.